test/integration/prometheus/uid_verification_test.go

```diff
@@ -0,0 +1,255 @@
+		// Resource naming: namespace-scoped resources use "namespace/name" format,
+		// cluster-scoped resources (nodes, PVs, namespaces) use just the name
+		switch metric {
+		// Deployment metrics
+		case "deployment_match_labels", "kube_deployment_spec_replicas",
+			"kube_deployment_status_replicas_available":
+			if item.Metric.Deployment != "" && item.Metric.Namespace != "" {
+				resourceName = fmt.Sprintf("%s/%s", item.Metric.Namespace, item.Metric.Deployment) // namespace-scoped
+			}
+		// Job metrics
+		case "kube_job_status_failed":
+			if item.Metric.JobName != "" && item.Metric.Namespace != "" {
+				resourceName = fmt.Sprintf("%s/%s", item.Metric.Namespace, item.Metric.JobName) // namespace-scoped
+			}
+		// Namespace metrics
+		case "kube_namespace_annotations", "kube_namespace_labels":
+			if item.Metric.Namespace != "" {
+				resourceName = item.Metric.Namespace // cluster-scoped
+			}
+		// Node metrics
+		case "kube_node_status_capacity", "kube_node_status_capacity_memory_bytes",
+			"kube_node_status_capacity_cpu_cores", "kube_node_status_allocatable",
+			"kube_node_status_allocatable_cpu_cores", "kube_node_status_allocatable_memory_bytes",
+			"kube_node_labels", "kube_node_status_condition":
+			if item.Metric.Node != "" {
+				resourceName = item.Metric.Node // cluster-scoped
+			}
+		// Pod metrics
+		case "kube_pod_labels", "kube_pod_owner", "kube_pod_container_status_running",
+			"kube_pod_container_status_terminated_reason", "kube_pod_container_status_restarts_total",
+			"kube_pod_container_resource_requests", "kube_pod_container_resource_limits",
+			"kube_pod_container_resource_limits_cpu_cores", "kube_pod_container_resource_limits_memory_bytes",
+			"kube_pod_status_phase":
+			if item.Metric.Pod != "" && item.Metric.Namespace != "" {
+				resourceName = fmt.Sprintf("%s/%s", item.Metric.Namespace, item.Metric.Pod) // namespace-scoped
+			}
+		// PVC metrics
+		case "kube_persistentvolumeclaim_resource_requests_storage_bytes",
+			"kube_persistentvolumeclaim_info":
+			if item.Metric.PersistentVolumeClaim != "" && item.Metric.Namespace != "" {
+				resourceName = fmt.Sprintf("%s/%s", item.Metric.Namespace, item.Metric.PersistentVolumeClaim) // namespace-scoped
+			}
+		// PV metrics
+		case "kube_persistentvolume_capacity_bytes", "kube_persistentvolume_status_phase",
+			"kubecost_pv_info":
+			if item.Metric.PersistentVolume != "" {
+				resourceName = item.Metric.PersistentVolume // cluster-scoped
+			}
+		// Service metrics
+		case "service_selector_labels":
+			if item.Metric.Service != "" && item.Metric.Namespace != "" {
+				resourceName = fmt.Sprintf("%s/%s", item.Metric.Namespace, item.Metric.Service) // namespace-scoped
+			}
+		// StatefulSet metrics
+		case "statefulSet_match_labels":
+			if item.Metric.StatefulSet != "" && item.Metric.Namespace != "" {
+				resourceName = fmt.Sprintf("%s/%s", item.Metric.Namespace, item.Metric.StatefulSet) // namespace-scoped
```

Comment:
this logic feels like it would take a lot of work to maintain. Yes, the upside here is that the names are nicely formatted, but what if you just like called ToString on the item.Metric? yes may look junkier but then would not need large case statement

Reply:
The case block looks repetitive. How about refactoring the case block to a single function?